### ISSUE TITLE
Support for [SQL Database Sync to other databases]

### ISSUE THREAD

**Issue by SadokLadhari** [hashicorp/terraform-provider-azurerm#6425]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Support for Sync to other databases.

### Affected Resource(s)

* azurerm_sql_database

### References

* [Sync Members - Create Or Update](https://docs.microsoft.com/en-us/rest/api/sql/syncmembers/createorupdate?view=servicefabric-5.7.198)
* [Sync Groups - Create Or Update](https://docs.microsoft.com/en-us/rest/api/sql/syncgroups/createorupdate?view=servicefabric-5.7.198)
* [Sync Agents - Create Or Update](https://docs.microsoft.com/en-us/rest/api/sql/syncagents/createorupdate?view=servicefabric-5.7.198)

**Comment by manicminer** [comment]
Blocked by https://github.com/Azure/azure-rest-api-specs/issues/11061

**Comment by katbyte** [comment]
It appears this issue relates to `azurerm_sql_database`, which was removed in v4.0 in favour of `azurerm_mssql_database` (see the [v4.0 upgrade guide](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/4.0-upgrade-guide#azurerm_sql_database) for details), as such we are going to close this issue out as it no longer applies to the current version of the provider.

If this is still relevant on the latest version of the provider (v5.x) please open a new issue against `azurerm_mssql_database` referencing this one. Thank you!
